### ISSUE TITLE
[GlobalISel] Improve the debug-ability of the match tables by adding the index as comment in the table

### ISSUE THREAD

**Issue by qcolombet** [llvm/llvm-project#119177]
When looking at the debug output of GlobalISel for the instruction selection phase, one can see something like:
```
10: GIM_SwitchOpcode(MIs[0], [51, 795), Default=357167, JumpTable...) // Got=51
2997: GIM_SwitchType(MIs[0]->getOperand(0), [2, 24), Default=21984, JumpTable...) // Got=s32
3086: Begin try-block
3093: GIM_CheckType(MIs[0]->getOperand(1), TypeID=2)
3096: GIM_CheckType(MIs[0]->getOperand(2), TypeID=2)
```

The number on the left are indices in the match table produced by the TableGen backend.
However, these numbers are not printed as comment in the `.inc` file making it difficult to exactly follow which entry of the table is being executed (you have a proxy information with the `@` comment, but those are sparse.)

The goal of this issue is to improve the debug-ability of the match table by adding the index of the table on each line.
E.g., go from
```
const uint8_t *AArch64InstructionSelector::getMatchTable() const {
  constexpr static uint8_t MatchTable0[] = {
    GIM_SwitchOpcode, /*MI*/0, /*[*/GIMT_Encode2(51), GIMT_Encode2(795), /*)*//*default:*//*Label 146*/ GIMT_Encode4(357167),
    /*TargetOpcode::G_ADD*//*Label 0*/ GIMT_Encode4(2986),
```
To
```
const uint8_t *AArch64InstructionSelector::getMatchTable() const {
  constexpr static uint8_t MatchTable0[] = {
/* 0 */    GIM_SwitchOpcode, /*MI*/0, /*[*/GIMT_Encode2(51), GIMT_Encode2(795), /*)*//*default:*//*Label 146*/ GIMT_Encode4(357167),
/* 10*/    /*TargetOpcode::G_ADD*//*Label 0*/ GIMT_Encode4(2986),
```

In other words something similar to SDISel:
```
  static const unsigned char MatcherTable[] = {
/*     0*/ OPC_SwitchOpcode /*448 cases */, 54|128,114|128,1/*31030*/, TARGET_VAL(ISD::EXTRACT_VECTOR_ELT),// ->31036
/*     6*/  OPC_Scope, 103|128,96|128,1/*28775*/, /*->28785*/ // 4 children in Scope
/*    10*/   OPC_MoveChild0,
```
